packages/fe-mockserver-core/src/router/batchRouter.ts

```diff
@@ -7,26 +7,104 @@ import type { Batch, BatchPart } from './batchParser';
 import { BatchContent, getBoundary, parseBatch } from './batchParser';
 import type { IncomingMessageWithTenant } from './serviceRouter';
 
+type Error412 = Record<string, ErrorResponse>;
+
+type ErrorResponse = {
+    code: string;
+    message: string;
+    severity?: string;
+    details: ErrorDetails[];
+} & Record<string, unknown>;
+
+type ErrorDetails = {
+    code: string;
+    message: string;
+    severity?: string;
+} & Record<string, unknown>;
+
+/**
+ * Returns the result whether isRequest part of changeset or not.
+ * @param part
+ * @returns {boolean}
+ */
 export function isPartChangeSet(part: BatchPart | Batch): part is Batch {
     return (part as Batch).isChangeSet;
 }
+
+/**
+ * Returns whether the response returns a 412 or not.
+ * @param partRequest
+ * @returns {boolean}
+ */
+function isResponse412(partRequest: ODataRequest) {
```

Comment:
this is a one liner that is called once, please donc create a function for it

Reply:
done